core/src/avm2/domain.rs

```diff
@@ -245,6 +245,9 @@ impl<'gc> Domain<'gc> {
     }
 
     pub fn export_class(&self, class: GcCell<'gc, Class<'gc>>, mc: MutationContext<'gc, '_>) {
+        if self.0.read().classes.contains_key(class.read().name()) {
+            return;
```

Comment:
I do not know ActionScript, and I do not quite understand this code either (not because it is unclear, the code seems clear, but because I do not know much about this part of the codebase), but, if I had to guess, if we "include" a class/type (like `MyDuplicate`), and we then "include" it some other way (using for instance `Loader.load()`), we do not want to "include" what is the same class again, since it has already been "included", correct? If so, the change seems good. Though, is `name()` qualified with a package path (if that makes sense)? Like, if I had two different types `package1.MyDuplicate` and `package2.MyDuplicate`, would it handle that correctly? But I do not know ActionScript and I do not quite understand this part of the codebase.

Reply:
You can see how the fix is relevant in the recently added test. Nothing actionable in this thread, so resolving.

Reply:
That is not actually what I am asking about, sorry if I am being unclear. I did see the test, and I am referring to it in the comment. What I first asked about was if my understanding of the changes (and the test) is correct, and then I asked if a certain corner case is handled correctly. Which I am still in doubt about - do you know if `name()`  is qualified with a package path (if that makes sense)?